Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h

```diff
@@ -2039,16 +2039,17 @@ class Surface_mesh
     return os.good();
   }
 
-
   /// \relates Surface_mesh
-  /// Inserts the surface mesh in an output stream in Ascii OFF format. 
-  /// Only the \em point property is inserted in the stream.
-  /// \pre `operator<<(std::ostream&,const P&)` must be defined.
-
+  /// 
+  /// This operator calls `write_off(std::istream& , CGAL::Surface_mesh)`.
    template <typename P>
   std::ostream& operator<<(std::ostream& os, const Surface_mesh<P>& sm)
   {
-    write_off(os, sm);
+    if(file_format(os) == IO::OFF){
+      write_off(os, sm);
+    } else {
+      std::cerr << "For Surface_mesh we only support the OFF file format" << std::endl;
+    }
```

Comment:
That additional state stored in the I/O stream dynamic flags seems wrong. That is the sort of thing that forces to chain several `if()` or a big `switch()` in every I/O functions.

I have **already** talked about that very well designed API `CGAL::oformat`/`CGAL::iformat`, that replaced that dynamic dispatching by a (static) tag dispatching.

I case one call a function with a variant that does not exist, using a tag dispatching, there would be a compilation error instead of that dynamic message on `cerr`. I really think that the tag dispatching is a lot better for any use-case.